File: rev_modules/base/models/rev_menu.js

var rev = require('../../../rev-server');

exports.RevMenu = rev.RevModel.extend({
	_name: 'rev.menu',
	_description: 'Rev App Menu',
	
	_columns: {
		name: 'name field def...',
		parent_id: 'parent field def...',
	}
});

File: rev_modules/rev_app/models/RevMenu.py

from rev.core import models, fields
from rev.core.translations import translate as _

class RevMenu(models.RevModel):

    _description = 'Rev App UI Menu'
    
    name = fields.TextField(_('Menu Name'))
    parent = fields.RecordLinkField(_('Parent Menu'), 'RevMenu')
    action = fields.RecordLinkField(_('Action'), 'RevAction', required=False)
    icon = fields.TextField(_('Menu Icon'), required=False)
    sequence = fields.IntegerField(_('Menu Sequence'))
